### ISSUE TITLE
about the mesh vertices format(xyz or uvd)

### ISSUE THREAD

**Issue by dumyy** [arielai/youtube_3d_hands#13]
Thanks for sharing the great dataset and relative work.
I have some dobuts abut the mesh 3d formats.  In the `load_db.py`, the function just plots the projected mesh points using the mesh vertices' `x and y` ,e.g.  `plt.plot(vertices[:, 0], vertices[:, 1], 'o', color='green', markersize=1)` in [line 56](https://github.com/arielai/youtube_3d_hands/blob/38b6de0571a77729c46be4cd446f26df67d320a6/load_db.py#L56), and the paper also indicates that
 "meshes in the image coordinate system is better than pretraining in the canonical frame and estimating camera parameters." 
So, did the json of the label about the dataset just use the data format( u,v d),d is the scaled depth, instead of format(x,y,z) in camera C.S. ?  If so, how about the output of the net? did it also observe the same format(uvd)? And could you please share some method about normalization of mesh's gt while training the net?
Thank you again!

**Comment by quyanqiu** [comment]
Hi,I am not the author of this paper,but I would like to share some my understanding.
Actually,Yes, you are right,The output of the network is generally speaking '(u,v,d)'.I wanna talk about the GT mesh generation step,e.g the “itterative fitting” method in the paper ，In fact this optimization based method did generate 3D hand mesh in camera C.S，But for better convergence for spiralNet ,the Arthur align these vertices to image C.S,which in your words '(u,v,d)'，
And this is why spiral-cov-net working because of this strict constraint of "point(x,y)-to-point(u,v,d)",so you can regard these approach as a "dense 3D hand keypoints estimation task".some of the other scholars work such as https://github.com/mks0601/I2L-MeshNet_RELEASE also did the same thing

As for normalization of mesh‘s gt,you can use following code:
 `
    train_db_np, val_db_np = np.array(data_splits['train'], dtype=np.float32), np.array(data_splits['val'], dtype=np.float32)

    mean_points = np.mean(train_db_np, axis=0)
    std_points = np.std(train_db_np, axis=0)

    # Normalize vertices.
    train_db_np = (train_db_np - mean_points) / std_points
    val_db_np = (val_db_np - mean_points) / std_points
`

This is also @dkulon ‘s work

**Comment by dumyy** [comment]
@quyanqiu In fact, I have also another confusion. If I have got the mesh output, or just use the gt of mesh, how to project the points to 3d format ( in camera C.S. or just root-relative C.S.)?

**Comment by quyanqiu** [comment]
@dumyy,Actually the gt of mesh already in 3d format（Although we call it 'u v d'）,you could use meshlab to visualize these vertices,just like below:
![image](https://user-images.githubusercontent.com/11349003/96824992-bae74000-1462-11eb-89a0-71f2d0c78fe5.png)
(BTW the faces are from mano website)
